### PR TITLE
[coreSNTP] Add doxygen check for CI

### PR DESCRIPTION
**Pull request by gshvang**
This PR updated ci.yml file to add the Doxygen check and add the Doxygen files needed to build Doxygen.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -144,3 +144,11 @@ jobs:
           path: ./
           exclude-dirs: cbmc
           include-file-types: .c,.h,.dox
+  doxygen:
+    runs-on: ubuntu-20.04
```
Comment (aggarw13) [inline .github/workflows/ci.yml:148]: We can use `ubuntu-latest` instead as GitHub had transitioned to use Ubuntu 20.04 in the beginning of the year
```suggestion
    runs-on: ubuntu-latest
```

File: docs/doxygen/config.doxyfile [hunk 2]
```diff
@@ -0,0 +1,2005 @@
...
+# could be handy for archiving the generated documentation or if some version
+# control system is used.
+
+PROJECT_NUMBER         = "v1.0.0"
+
+# Using the PROJECT_BRIEF tag one can provide an optional one line description
+# for a project that appears at the top of each page and should give viewer a
+# quick idea about the purpose of the project. Keep the description short.
+
+PROJECT_BRIEF          = "Client library to use Simple Network Time Protocol (SNTP) to synchronize device clocks with internet time"
```
Comment (aggarw13) [inline docs/doxygen/config.doxyfile:47]:
```suggestion
PROJECT_BRIEF          = "Client library for synchronizing device time with internet time using Simple Network Time Protocol (SNTP)"
```

File: docs/doxygen/include/size_table.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<table>
+    <tr>
+        <td colspan="3"><center><b>Code Size of coreJSON</b></center></td>
```
Comment (aggarw13) [inline docs/doxygen/include/size_table.html:3]: This file is incomplete.. Can you also add the check for memory statistics?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -144,3 +144,26 @@ jobs:
...
+  memory_statistics:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+            submodules: 'recursive'
+      - name: Install Python3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7.10'
```
Comment (aggarw13) [inline .github/workflows/ci.yml:164]:
```suggestion
          python-version: '3.7.x'
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -144,3 +144,26 @@ jobs:
...
+      - name: Run doxygen build
+        uses: FreeRTOS/CI-CD-Github-Actions/doxygen@main
+        with:
+          path: ./
+  memory_statistics:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+            submodules: 'recursive'
```
Comment (aggarw13) [inline .github/workflows/ci.yml:160]: I don't think recursive clone should be required for running the action
```suggestion
```

File: .github/workflows/memory_statistics.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+name: Memory statistics
```
Comment (aggarw13) [inline .github/workflows/memory_statistics.yml:1]: This separate workflow of generating memory statistic values is not required as the PR check ensures that the memory statistics table is always up-to-date.

File: docs/doxygen/config.doxyfile [hunk 7]
```diff
@@ -0,0 +1,2005 @@
...
+# the path). If a tag file is not located in the directory in which doxygen is
+# run, you must also specify the path to the tagfile here.
+
+TAGFILES               =
+
+# When a file name is specified after GENERATE_TAGFILE, doxygen will create a
+# tag file that is based on the input files it reads. See section "Linking to
+# external documentation" for more information about the usage of tag files.
+
+GENERATE_TAGFILE       = docs/doxygen/output/json.tag
```
Comment (aggarw13) [inline docs/doxygen/config.doxyfile:1731]:
```suggestion
GENERATE_TAGFILE       = docs/doxygen/output/sntp.tag
```

**Review by aggarw13 — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**